Semana_19/Aula_20012020/src/aula-paginacao/business/gateway/f4User/f4UserGateway.ts

```diff
@@ -0,0 +1,25 @@
+import { F4User } from "../../entities/f4User";
+
+export interface F4CreateUserGateway {
+  createUser(user: F4User): Promise<void>;
+}
+
+export interface F4GetAllUsersGateway {
+  getAllUsers(): Promise<F4User[]>;
+}
+
+export interface F4GetUsersByTypeGateway {
+  getUsersByType(type: string): Promise<F4User[]>;
+}
+
+export interface F4GetOrderedUsersGateway {
+  getOrderedUsers(orderBy: string, orderType: string): Promise<F4User[]>;
```

Comment:
Tente não deixar arquivos e interfaces que você não for utilizar. Entendo que, as vezes, fica mais fácil pegar o exemplo da aula, mas lembre-se de adaptá-lo ao exercício de tarde!